From 1ea30dc27488718966d38ca8cad97da25f8a840e Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Tue, 27 Nov 2012 11:42:39 -0800
Subject: select.mk: make path to ccache externally visible as CCACHE_BIN

Some other Makefiles can make use of this information if they
need to invoke it directly.

Change-Id: I31dbff712f71bed0db85f8de16ad005d8d1c9578
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
---
 core/combo/select.mk |   15 +++++++--------
 1 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/core/combo/select.mk b/core/combo/select.mk
index e485d00..b303bad 100644
--- a/core/combo/select.mk
+++ b/core/combo/select.mk
@@ -83,17 +83,16 @@ ifneq ($(USE_CCACHE),)
   ifeq ($(HOST_OS)-$(BUILD_OS),windows-linux)
     CCACHE_HOST_TAG := linux-$(BUILD_ARCH)
   endif
-  ccache := prebuilts/misc/$(CCACHE_HOST_TAG)/ccache/ccache
+  CCACHE_BIN := prebuilts/misc/$(CCACHE_HOST_TAG)/ccache/ccache
   # Check that the executable is here.
-  ccache := $(strip $(wildcard $(ccache)))
-  ifdef ccache
+  CCACHE_BIN := $(strip $(wildcard $(CCACHE_BIN)))
+  ifdef CCACHE_BIN
     # prepend ccache if necessary
-    ifneq ($(ccache),$(firstword $($(combo_target)CC)))
-      $(combo_target)CC := $(ccache) $($(combo_target)CC)
+    ifneq ($(CCACHE_BIN),$(firstword $($(combo_target)CC)))
+      $(combo_target)CC := $(CCACHE_BIN) $($(combo_target)CC)
     endif
-    ifneq ($(ccache),$(firstword $($(combo_target)CXX)))
-      $(combo_target)CXX := $(ccache) $($(combo_target)CXX)
+    ifneq ($(CCACHE_BIN),$(firstword $($(combo_target)CXX)))
+      $(combo_target)CXX := $(CCACHE_BIN) $($(combo_target)CXX)
     endif
-    ccache =
   endif
 endif
-- 
1.7.4.1

